services/src/main/java/org/fao/geonet/api/users/MeApi.java

```diff
@@ -23,71 +23,125 @@
+    public class MeApi {
+
+        @ApiOperation(
+            value = "Get information about me",
+            notes = "If not authenticated, return status 204 (NO_CONTENT), " +
+                "else return basic user information. This operation is usually used to " +
+                "know if current user is authenticated or not." +
+                "It returns also info about groups and profiles.",
+            nickname = "getMe")
+        @RequestMapping(
+            produces = MediaType.APPLICATION_JSON_VALUE,
+            method = RequestMethod.GET)
+        @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Authenticated. Return user details."),
+            @ApiResponse(code = 204, message = "Not authenticated.")
+        })
+        @ResponseBody
+        public ResponseEntity<MeResponse> getMe(
+            @ApiIgnore
+            @ApiParam(
+                hidden = true
+            )
+            HttpSession session
+        ) throws Exception {
+
         UserSession userSession = ApiUtils.getUserSession(session);
-        return userSession.isAuthenticated() ?
-            new ResponseEntity<>(new MeResponse()
-                .setId(userSession.getUserId())
-                .setProfile(userSession.getProfile().name())
-                .setUsername(userSession.getUsername())
-                .setName(userSession.getName())
-                .setSurname(userSession.getSurname())
-                .setEmail(userSession.getEmailAddr())
-                .setOrganisation(userSession.getOrganisation()),
-                    HttpStatus.OK) :
-            new ResponseEntity<>(NO_CONTENT);
+
+        if (userSession.isAuthenticated()) {
+
+            MeResponse myInfos = new MeResponse().setId(userSession.getUserId()).setProfile(userSession.getProfile().name())
+                    .setUsername(userSession.getUsername()).setName(userSession.getName()).setSurname(userSession.getSurname())
+                    .setEmail(userSession.getEmailAddr()).setOrganisation(userSession.getOrganisation())
+                    .setAdmin(userSession.getProfile().equals(Profile.Administrator))
+                    .setGroupsWithRegisteredUser(getGroups(userSession, Profile.RegisteredUser))
+                    .setGroupsWithEditor(getGroups(userSession, Profile.Editor))
+                    .setGroupsWithReviewer(getGroups(userSession, Profile.Reviewer))
+                    .setGroupsWithUserAdmin(getGroups(userSession, Profile.UserAdmin));
+
+            return new ResponseEntity<>(myInfos, HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>(NO_CONTENT);
+    }
+
+    /**
+     * Retrieves a user's groups.
+     *
+     * @param includingSystemGroups if true, also returns the system groups ('GUEST', 'intranet', 'all')
+     * @param all if true returns all the groups, even those the user doesn't belongs to
+     */
+    private List<Integer> getGroups(UserSession session, Profile profile) throws SQLException {
```

Comment:
Can't be obtained all the user groups once and filter by profile in Java code?

Reply:
I followed the other suggestion, now only the GroupIds are extracted, so it's not possible to filter